packages/open-next/src/core/routing/middleware.ts

```diff
@@ -151,17 +152,16 @@ export async function handleMiddleware(
       isExternalRewrite = true;
     } else {
       const rewriteUrlObject = new URL(rewriteUrl);
+      // If we have a rewrite, search params from the rewrite URL should be used and override the original search params
 
-      // Reset the query params if the middleware is a rewrite
-      middlewareQueryString = middlewareQueryString.__nextDataReq
-        ? {
-            __nextDataReq: middlewareQueryString.__nextDataReq,
-          }
-        : {};
+      middlewareQueryString = getQueryFromSearchParams(
```

Comment:
Could you rename `middlewareQueryString` to `middlewareQuery` instead as it is not a string (anymore?)